terraform_files/vsphere/main.tf

```diff
@@ -0,0 +1,93 @@
+provider "vsphere" {
+  user                 = var.vsphere_username
+  password             = var.vsphere_password
+  vsphere_server       = var.vsphere_vcenter
+  allow_unverified_ssl = true
+}
+
+data "vsphere_datacenter" "datacenter" {
+  name = var.vsphere_datacenter
+}
+
+data "vsphere_compute_cluster" "cluster" {
+  name          = var.vsphere_cluster
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_datastore" "datastore" {
+  name          = var.vsphere_datastore
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_network" "network" {
+  name          = var.vsphere_network
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+resource "vsphere_tag_category" "category" {
+  name        = "openshift-${var.cluster_name}"
+  description = "Added by openshift-install do not remove"
+  cardinality = "SINGLE"
+
+  associable_types = [
+    "VirtualMachine",
+    "ResourcePool",
+    "Folder"
+  ]
+}
+
+resource "vsphere_tag" "tag" {
+  name        = var.cluster_name
+  category_id = vsphere_tag_category.category.id
+  description = "Added by openshift-install do not remove"
+}
+
+resource "vsphere_folder" "folder" {
+  count = 1
+  path          = var.cluster_name
+  type          = "vm"
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+  tags          = [vsphere_tag.tag.id]
+}
+
+resource "vsphere_file" "ISO_UPLOAD" {
+  datacenter       = var.vsphere_datacenter
+  datastore        = var.vsphere_datastore
+  source_file      = var.iso_download_path
+  destination_file = "ISOs/${basename(var.iso_download_path)}"
```

Comment:
The `iso_download_path` should be absolute

Reply:
This property is the path on the vsphere itself